File: web-app/src/app/services/types/Template.ts
import { TemplateCategory } from "./TemplateCategory";
import { TemplatePlaceholder } from "./TemplatePlaceholder";

export type Template = {
  Id: number;
  CategoryId: number;
  Name: string;
  Subject: string;
  Body: string;
  CreatedAt: Date;
  UpdatedAt: Date;
  Category: TemplateCategory;
  TemplatePlaceholders: TemplatePlaceholder[];
};

File: Placements.WebApi/Placement/Models/Template.cs
using System;
using System.Collections.Generic;

namespace Placements.WebApi.Placement.Models;

public partial class Template
{
    public long Id { get; set; }

    public int? CategoryId { get; set; }

    public string? Name { get; set; }

    public string? Subject { get; set; }

    public string? Body { get; set; }

    public DateTime? CreatedAt { get; set; }

    public DateTime? UpdatedAt { get; set; }

    public virtual TemplateCategory? Category { get; set; }

    public virtual ICollection<TemplatePlaceholder> TemplatePlaceholders { get; set; } = new List<TemplatePlaceholder>();
}
